src/integrationTest/java/uk/gov/hmcts/reform/bulkscan/payment/processor/client/payhub/PayHubClientTest.java

```diff
@@ -0,0 +1,151 @@
+package uk.gov.hmcts.reform.bulkscan.payment.processor.client.payhub;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.tomakehurst.wiremock.client.ResponseDefinitionBuilder;
+import com.google.common.collect.ImmutableList;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cloud.contract.wiremock.AutoConfigureWireMock;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.junit4.SpringRunner;
+import uk.gov.hmcts.reform.bulkscan.payment.processor.client.payhub.request.PaymentRequest;
+import uk.gov.hmcts.reform.bulkscan.payment.processor.client.payhub.response.PaymentResult;
+import uk.gov.hmcts.reform.bulkscan.payment.processor.config.IntegrationTest;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.badRequest;
+import static com.github.tomakehurst.wiremock.client.WireMock.equalToJson;
+import static com.github.tomakehurst.wiremock.client.WireMock.okJson;
+import static com.github.tomakehurst.wiremock.client.WireMock.post;
+import static com.github.tomakehurst.wiremock.client.WireMock.serverError;
+import static com.github.tomakehurst.wiremock.client.WireMock.stubFor;
+import static com.google.common.net.HttpHeaders.CONTENT_TYPE;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.catchThrowable;
+import static org.springframework.http.HttpStatus.BAD_REQUEST;
+import static org.springframework.http.HttpStatus.INTERNAL_SERVER_ERROR;
+
+@AutoConfigureWireMock
+@IntegrationTest
+@RunWith(SpringRunner.class)
+public class PayHubClientTest {
+
+    @Autowired
+    private PayHubClient client;
+
+    @Autowired
+    private ObjectMapper mapper;
+
+    @Test
+    public void should_return_Ok_when_everything_is_ok_with_request() throws JsonProcessingException {
+        // given
+        PaymentResult response = new PaymentResult(ImmutableList.of("xxxyyyzzz", "zzzyyyxxx"));
+        stubWithRequestAndResponse(mapper.writeValueAsString(getPaymentRequest()),
```

Comment:
Is there a reason to use Wiremock over Mockito ?

Reply:
used for integration test